kubernetes_state/check.py

```diff
@@ -172,6 +173,8 @@ def check(self, instance):
         self.job_succeeded_count = defaultdict(int)
         self.job_failed_count = defaultdict(int)
 
+        self.whitelisted_reasons = {"waiting":["ErrImagePull"],"terminated":["OOMKilled","ContainerCannotRun","Error"]}
```

Comment:
They'll be created at every run. you can put them as module constants at the top, like `METRIC_TYPES`. You should assign two separate arrays instead of a dict for performance.